app/src/main/java/jaelson/wagner/silva/upcomingmovies/di/module/NetworkModule.java

```diff
@@ -0,0 +1,78 @@
+package jaelson.wagner.silva.upcomingmovies.di.module;
+
+import com.facebook.stetho.okhttp3.StethoInterceptor;
+import dagger.Module;
+import dagger.Provides;
+import jaelson.wagner.silva.upcomingmovies.ApplicationConfiguration;
+import jaelson.wagner.silva.upcomingmovies.BuildConfig;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Converter;
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
+import retrofit2.converter.moshi.MoshiConverterFactory;
+
+import javax.annotation.Nullable;
+import javax.inject.Singleton;
+
+/**
+ * Created by jaelsonwagner@gmail.com.
+ */
+
+@Module
+public class NetworkModule {
+
+  @Provides
+  @Nullable
+  HttpLoggingInterceptor provideHttpLoggingInterceptor() {
+    if (BuildConfig.DEBUG) {
+      return new HttpLoggingInterceptor((String message) -> {
+        // we can emit event here to a client listener handle that data, like persist
+        // requests
+        // into a database to future query, i.e., a debug drawer screen.
+      }).setLevel(HttpLoggingInterceptor.Level.BODY);
+    }
+
+    return null;
+  }
+
+  @Provides
+  @Nullable
+  Interceptor provideInterceptor() {
+    return BuildConfig.DEBUG ? new StethoInterceptor() : null;
+  }
+
+  @Provides
+  OkHttpClient provideOkHttpClient(@Nullable HttpLoggingInterceptor httpLoggingInterceptor,
+                                   @Nullable Interceptor networkInterceptor) {
+
+    OkHttpClient.Builder builder = new OkHttpClient.Builder();
+
+    if (httpLoggingInterceptor != null) {
+      builder.addInterceptor(httpLoggingInterceptor);
+    }
+
+    if (networkInterceptor != null) {
+      builder.addNetworkInterceptor(networkInterceptor);
+    }
+
+    return builder.build();
+  }
+
+  @Provides
+  Converter.Factory provideConverterFactory() {
+    return MoshiConverterFactory.create().withNullSerialization().failOnUnknown();
```

Comment:
`failOnUnknown()` broke the app for me, it seems the API changed recently... Why so rigid?

Reply:
Well, I have found some API bugs while developing, so, I applied that configuration to try get some failure while developing. But, I should  removed it before delivery.